cmd/run.go

```diff
@@ -17,13 +17,13 @@ func runSemangit(cmd *cobra.Command, args []string) error {
 	if conf.SrcRevision != gitrepo.RevisionNone {
 		repoManager.Checkout(conf.SrcRevision)
 	}
-
-	// TODO: Check if r.versionAnalyzer.ChangeNeedsVersionUpdate
 	srcVersion := utils.GetResultOrPanic(versionAnalyzer.ReadVersion(conf.RepoDir, conf.GetCurrentVersionAnalyzerArgumentValues()))
 	repoManager.Checkout(conf.DestRevision)
 	toVersion := utils.GetResultOrPanic(versionAnalyzer.ReadVersion(conf.RepoDir, conf.GetCurrentVersionAnalyzerArgumentValues()))
-
-	fmt.Printf("TODO: Compare versions! From Version: %s To Version: %s\n", srcVersion, toVersion)
-	fmt.Printf("Compare result: %d\n", versionAnalyzer.CompareVersions(srcVersion, toVersion))
+	changedFiles := repoManager.ListChangedFiles(conf.SrcRevision, conf.DestRevision)
+	needsUpdate := versionAnalyzer.ChangeNeedsVersionUpdate(changedFiles, conf.GetCurrentVersionAnalyzerArgumentValues())
+	if needsUpdate && versionAnalyzer.CompareVersions(srcVersion, toVersion) == 0 {
```

Comment:
That's rare, but it's possible to see the version decrease!
```suggestion
	if needsUpdate && versionAnalyzer.CompareVersions(srcVersion, toVersion) >= 0 {
```

Reply:
Doesn't it have to be `<=`?

Reply:
`>= 0` means `srcVersion >= destVersion`. Doesn't it?

Reply:
Yeah, and we have to raise an error when `srcVersion <= destVersion`. If I'm right, `destVersion` will be _master_ in most cases and other branches must have a greater or equal version to it, otherwise, we have to raise an error.

Reply:
No, it's vice versa. In our literature, the project is updating from `src` to `dest`; e.g. it's changing from what is at `master` (src) to `new-feature` (dest).